nourish-team/nourish-backend
# Setup Docker for PostgreSQL database

# Description

I have created a Docker compose file that creates an image of our PostgreSQL database. This means that we don't have to create a local database anymore and we directly can test on the PostgreSQL image.

# Closes issue(s)

Resolves #71 

# How to test

First you need to install Docker Engine to be able to run the PostgreSQL container: https://docs.docker.com/engine/install/

Make sure that your `DATABASE_URL` in your . env matches the docker environment. 

To run the compose file and migrations run the command:
`npm run db:dockerup`

After you are done using the the container run:
`npm run db:dockerdown`

Here are the following steps how to test:

1. Start the server `npm run dev`
2. Start docker `npm run dev:startdb`
3. Create a user: ``` {
    "username": "testuser",
    "email": "testuser@mail.com",
    "uid": "vdcksjhlvidhst"
}```
5. Run `npm run prisma:studio` and see if the user is created
6. To stop the container `npm run db:dockerdown`


# Changes include

1. Add a docker folder with a docker compose file
2. Add docker commands in package.json
3. Have to make changes to couple of files because of Eslint errors from previous PR.

# Checklist


# To do

- Update  README
